Tests/Realm.Tests/SourceGeneration/TestClasses/ManuallyGeneratedClass.cs

```diff
@@ -0,0 +1,270 @@
+            {
+                if (IsManaged && _propertyChanged == null)
+                {
+                    SubscribeForNotifications();
+                }
+
+                _propertyChanged += value;
+            }
+
+            remove
+            {
+                _propertyChanged -= value;
+
+                if (IsManaged &&
+                    _propertyChanged == null)
+                {
+                    UnsubscribeFromNotifications();
+                }
+            }
+        }
+
+        protected virtual void OnPropertyChanged(string propertyName)
+        {
+        }
+
+        protected void RaisePropertyChanged([CallerMemberName] string propertyName = null)
+        {
+            _propertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+            OnPropertyChanged(propertyName);
+        }
+
+        protected internal virtual void OnManaged()
+        {
+        }
+
+        private void SubscribeForNotifications()
+        {
+            _accessor.SubscribeForNotifications(RaisePropertyChanged);
+        }
+
+        private void UnsubscribeFromNotifications()
+        {
+            _accessor.UnsubscribeFromNotifications();
+        }
+    }
+}
+
+// Having a separate namespace allows to hide the implementation details better.
+namespace Realm.Generated
+{
+    internal class ManuallyGeneratedClassObjectHelper : IRealmObjectHelper
+    {
+        public void CopyToRealm(IRealmObjectBase instance, bool update, bool skipDefaults)
+        {
+            throw new InvalidOperationException("This method should not be called for source generated classes.");
+        }
+
+        public ManagedAccessor CreateAccessor() => new ManuallyGeneratedClassManagedAccessor();
+
+        public IRealmObjectBase CreateInstance()
+        {
+            return new ManualllyGeneratedClass();
+        }
+
+        public bool TryGetPrimaryKeyValue(IRealmObjectBase instance, out object value)
+        {
+            value = (instance.Accessor as IManuallyGeneratedClassAccessor).PrimaryKeyValue;
+            return true;
+        }
+    }
+
+    internal interface IManuallyGeneratedClassAccessor : IRealmAccessor
+    {
+        int IntValue { get; set; }
+
+        IList<int> ListValue { get; }
+
+        public string StringValue { get; set; }
+
+        public int PrimaryKeyValue { get; set; }
+    }
+
+    internal class ManuallyGeneratedClassManagedAccessor
+        : ManagedAccessor, IManuallyGeneratedClassAccessor
+    {
+        /** If we want to make this more efficient, we can use property indexes here.
+         * We should have all the info necessary to compute them during source generation (as we do in Realm.CreateRealmObjectMetadata).
+         */
+        public int IntValue
+        {
+            get => (int)GetValue("IntValue");
+            set => SetValue("IntValue", value);
+        }
+
+        public IList<int> ListValue
+        {
+            get => GetListValue<int>("ListValue");
```

Comment:
Previously, we would only create the collection once rather than for every access. I suggest we employ a similar strategy here as well. Something like:

```csharp
private Lazy<IList<int>> _listValue = new(() => GetListValue<int>("ListValue"));
public IList<int> ListValue => _listValue.Value;
```

Reply:
Both are using a backing field :P Whether the field is lazy or not, doesn't matter all that much, so whatever you prefer. The reason we previously used a backing field instead of a lazy was that we already had the backing field for the auto property on the object and constructing a lazy in IL is... fun 😫